cms/djangoapps/contentstore/views/tests/test_course_index.py

```diff
@@ -340,6 +343,80 @@ def _assert_settings_link_present(response):
         self.assertEqual(_get_release_date(response), get_default_time_display(self.course.start))
         _assert_settings_link_present(response)
 
+    def _create_test_data(self, course_module, create_blocks=False, publish=True, block_types=None):
+        """
+        Create data for test.
+        """
+        if create_blocks:
+            for block_type in block_types:
+                ItemFactory.create(
+                    parent_location=self.vertical.location,
+                    category=block_type,
+                    display_name='{} Problem'.format(block_type)
+                )
+
+            if not publish:
+                self.store.unpublish(self.vertical.location, self.user.id)
+
+        course_module.advanced_modules.extend(block_types)
+
+    def _verify_deprecated_info(self, course_id, advanced_modules, info, deprecated_block_types, publish=True):
+        """
+        Verify deprecated info.
+        """
+        expected_blocks = []
+        if publish:
+            for block_type in deprecated_block_types:
+                expected_blocks.append(
+                    [
+                        reverse_usage_url('container_handler', self.vertical.location),
+                        '{} Problem'.format(block_type)
+                    ]
+                )
+
+        self.assertEqual(info['block_types'], deprecated_block_types)
+        self.assertEqual(
+            info['block_types_enabled'],
+            any(component in advanced_modules for component in deprecated_block_types)
+        )
+        self.assertItemsEqual(info['blocks'], expected_blocks)
+        self.assertEqual(
+            info['advance_settings_url'],
+            reverse_course_url('advanced_settings_handler', course_id)
+        )
+
+    @ddt.data(
+        {'publish': True},
+        {'publish': False},
+    )
+    @ddt.unpack
+    def test_verify_deprecated_warning_message_with_single_feature(self, publish):
+        """
+        Verify deprecated warning info for single deprecated feature.
+        """
+        block_types = settings.DEPRECATED_BLOCK_TYPES
+        course_module = modulestore().get_item(self.course.location)
+        self._create_test_data(course_module, create_blocks=True, block_types=block_types, publish=publish)
+        info = _deprecated_blocks_info(course_module, block_types)
+        self._verify_deprecated_info(
+            course_module.id,
+            course_module.advanced_modules,
+            info,
+            block_types,
+            publish=publish
+        )
+
+    def test_verify_deprecated_warning_message_with_multiple_features(self):
```

Comment:
Unless it's covered elsewhere, you need a negative test (i.e. that the warning message would NOT appear when there are no deprecated features on the course).